### PR TITLE
Add TODO Comment Checker and associated unit tests

### PR DESCRIPTION
**Pull request by rigoford**
Not sure what happened to the old pull request (#133). This one covers off a couple of suggestions by [HairyFotr](https://github.com/HairyFotr)

### CODE REVIEW

**Comment by rigoford** [comment]
Added a FIXME check as requested by [HairyFotr](https://github.com/HairyFotr)

File: src/main/resources/scalastyle_documentation.xml [hunk 1]
```diff
@@ -560,6 +560,45 @@ To bring consistency with how comments should be formatted, leave a space right
...
+   <checker level="warning" class="org.scalastyle.file.FixmeCommentChecker" enabled="true">
+     <parameters>
+       <parameter name="regex" type="string" default="(?i)(//|/\*|/\*\*|\*)\s?fixme(:?)\s+" />
+     </parameters>
+   </checker>
+ ]]>
+ </example-configuration>
+ </check>
+
+ <check id="fixme.comment">
```
Comment (matthewfarwell) [inline src/main/resources/scalastyle_documentation.xml]: This shouldn't be in twice

**Comment by matthewfarwell** [comment]
@HairyFotr It seems like overkill to have two classes. Please note that you can have the same class twice in the configuration, but with different parameters. Could you confirm that this would be acceptable?

File: src/main/scala/org/scalastyle/file/FixmeCommentChecker.scala [hunk 2]
```diff
@@ -0,0 +1,47 @@
+package org.scalastyle.file
+
+import org.scalastyle.file.FixmeCommentChecker._
+import org.scalastyle.{ColumnError, FileChecker, Lines, ScalastyleError}
+
+/**
+ * Checks for line comment style FIXMEs
+ */
+class FixmeCommentChecker extends FileChecker {
```
Comment (matthewfarwell) [inline src/main/scala/org/scalastyle/file/FixmeCommentChecker.scala]: This seems to be exactly the same code as the FixmeCommentChecker - you could use the same abstract class for them both. Also, I'm not sure that we need both classes anyway - see general comments.

**Comment by HairyFotr** [comment]
Cool, I didn't know the same class can be used multiple times. Yes, one class is enough, then.

**Comment by rigoford** [comment]
So I've created an abstract class for Todo and Fixme to extend. I wasn't sure about using just one class. How would the errorKey and id work?

**Comment by matthewfarwell** [comment]
Sorry it took so long to get back to this. Could you do a rebase and a squash for this please?

**Comment by rigoford** [comment]
Rebased and squashed as requested. It should be good.

**Comment by matthewfarwell** [comment]
I'm going to accept this PR, but I think we can do better for configuration - the regex seems very complicated and hard to get right. I'll change it to have one single string where you can specify TODO or FIXME etc, similarly to http://checkstyle.sourceforge.net/apidocs/com/puppycrawl/tools/checkstyle/checks/TodoCommentCheck.html

**Pull request merged**